go/server.go

```diff
@@ -113,40 +113,41 @@ func (s *Server) SetTokens(tokens map[string]string) {
 	s.config.Tokens = tokens
 }
 
-func (s *Server) authenticate(ctx context.Context) error {
+func (s *Server) authenticate(ctx context.Context) (context.Context, error) {
 	md, ok := metadata.FromIncomingContext(ctx)
 	if !ok {
-		return status.Error(codes.Unauthenticated, "missing request metadata")
+		return ctx, status.Error(codes.Unauthenticated, "missing request metadata")
```

Comment:
What's all the context plumbing for? I'm having trouble finding where it's actually used.

Reply:
The context plumbing allows us to pass the tag of the calling token down to specific methods, as it's supposed to be included in broadcasted events.